drivers/node/aws/aws.go

```diff
@@ -0,0 +1,119 @@
+package aws
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/Sirupsen/logrus"
+	aws_pkg "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/portworx/torpedo/drivers/node"
+)
+
+const (
+	// DriverName is the name of the aws driver
+	DriverName = "aws"
+)
+
+type aws struct {
+	node.Driver
+	session     *session.Session
+	credentials *credentials.Credentials
+	config      *aws_pkg.Config
+	region      string
+	svc         *ec2.EC2
+	instances   []*ec2.Instance
+}
+
+func (aws *aws) String() string {
+	return DriverName
+}
+
+func (aws *aws) Init(sched string) error {
+	sess := session.Must(session.NewSession())
+	aws.session = sess
+	creds := credentials.NewEnvCredentials()
+	aws.credentials = creds
+	aws.region = os.Getenv("AWS_REGION")
+	if aws.region == "" {
+		return fmt.Errorf("Env AWS_REGION not found")
+	}
+	config := &aws_pkg.Config{Region: aws_pkg.String(aws.region)}
+	config.WithCredentials(creds)
+	aws.config = config
+	svc := ec2.New(sess, config)
+	aws.svc = svc
+	instances := aws.getAllInstances()
+	aws.instances = instances
+	return nil
+}
+
+func (aws *aws) TestConnection(n node.Node, options node.TestConectionOpts) error {
+	return nil
+}
+
+func (aws *aws) RebootNode(n node.Node, options node.RebootNodeOpts) error {
```

Comment:
Change `aws` to `a` so we distinguish the instance and the struct name.

So, it'll be

`func (a *aws) ...`

Reply:
changed func(aws *aws) to func(a *aws)

Reply:
I don't see that updated here yet. Note that all of them need to have `func(a *aws)`

Reply:
not checked-in yet, I would check in soon